File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    answer = 0
    
    targets.sort(key=lambda x: x[1])
    curr = -1 # 가장 최근에 요격 미사일 발사 지점

    for target in targets:
        s, e = target
        if curr <= s:
            answer += 1
            curr = e
    
    return answer

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        
        Arrays.sort(targets, (o1, o2) -> (o1[1] - o2[1])); // e 기준 오름차순 정렬
        
        int curr = -1;
        
        for (int[] target: targets) {
            int s = target[0], e = target[1];
            if (curr < s) {
                curr = e - 1;
                answer++;
            }
        }
        
        return answer;
    }
}